### PR TITLE
Fire a change event when new background layer is changed

### PR DESCRIPTION
**Pull request by pgiraud**

### CODE REVIEW

**Comment by elemoine** [comment]
I realize that ngeo.js users won't be able to register listeners, because there's no exported register function they can use. One option is to inherit from ol.Observable instead of goog.events.EventTarget. With that ngeo.js users will be able to use the on and once functions. And users using Closure will still be able to use goog.events.listen.

**Comment by pgiraud** [comment]
Pull request updated.
It now inherits from `ol.Observable` + a `previous` property is provisioned.

File: externs/ngeox.js [hunk 1]
```diff
@@ -326,3 +326,14 @@ ngeox.SearchDirectiveListeners.prototype.selected;
  * @type {function(jQuery.event, Object, TypeaheadDataset)|undefined}
  */
 ngeox.SearchDirectiveListeners.prototype.autocompleted;
+
+
+/**
+ * @typedef {{previous: (ol.layer.Base|undefined)}}
```
Comment (elemoine) [inline externs/ngeox.js]: This should be interface.

File: src/services/backgroundlayermgr.js [hunk 2]
```diff
@@ -21,19 +21,67 @@
...
+  CHANGE: 'change'
+};
+
+
+
+/**
+ * @constructor
+ * @extends {goog.events.Event}
+ * @param {ngeo.BackgroundEventType} type Type.
+ * @param {ol.layer.Base|undefined=} opt_previous
```
Comment (elemoine) [inline src/services/backgroundlayermgr.js]: `@implements` missing.

**Comment by pgiraud** [comment]
Updated.

**Comment by elemoine** [comment]
Please merge.

**Pull request merged**